assets/js/Ioda/pages/dashboard/Dashboard.js

```diff
@@ -240,313 +238,293 @@ class Dashboard extends Component {
+            : "ne_10m_admin_1.regions";
+    if (mounted) {
+      console.log('mounted')
       getTopoAction(entityType)
-        .then((data) =>
-          topojson.feature(
-            data[entityType].topology,
-            data[entityType].topology.objects[topologyObjectName]
+          .then((data) =>
+              topojson.feature(
+                  data[entityType].topology,
+                  data[entityType].topology.objects[topologyObjectName]
+              )
           )
-        )
-        .then((data) =>
-          this.setState(
-            {
-              topoData: data,
-            },
-            this.getMapScores
-          )
-        );
+          .then((data) => {
+            console.log(data)
+            setTopoData(data)
+          })
     }
-  };
-  // function to manage when a user clicks a country in the map
-  handleEntityShapeClick = (entity) => {
-    const { navigate } = this.props;
+    console.log('328')
+  }
 
+  // function to manage when a user clicks a country in the map
+  function handleEntityShapeClick(entity) {
     // Use usercode for country, id for other types
     const entityCode =
-      this.state.activeTabType === country.type
-        ? entity.properties.usercode
-        : entity.properties.id;
-    let path = `/${this.state.activeTabType}/${entityCode}`;
+        activeTabType === country.type
+            ? entity.properties.usercode
+            : entity.properties.id;
+    let path = `/${activeTabType}/${entityCode}`;
 
     if (hasDateRangeInUrl()) {
-      path = `${path}/?from=${this.state.from}&until=${this.state.until}`;
+      path = `${path}/?from=${from}&until=${until}`;
     }
 
     navigate(path);
-  };
+  }
 
   // Event Time Series
-  getDataEvents(entityType) {
-    let until = this.state.until;
-    let from = this.state.from;
-    let attr = this.state.eventOrderByAttr;
-    let order = this.state.eventOrderByOrder;
-    let entities = this.state.summaryDataRaw
-      .map((entity) => {
-        // some entities don't return a code to be used in an api call, seem to default to '??' in that event
-        if (entity.entity.code !== "??") {
-          return entity.entity.code;
-        }
-      })
-      .toString();
-    this.props.getEventSignalsAction(
-      entityType,
-      entities,
-      from,
-      until,
-      attr,
-      order
+  function getDataEvents(entityType) {
+    let attr = eventOrderByAttr;
+    let order = eventOrderByOrder;
+    let entities = summaryDataRaw
+        .map((entity) => {
+          // some entities don't return a code to be used in an api call, seem to default to '??' in that event
+          if (entity.entity.code !== "??") {
+            return entity.entity.code;
+          }
+        })
+        .toString();
+    getEventSignalsAction(
+        entityType,
+        entities,
+        from,
+        until,
+        attr,
+        order
     );
   }
 
-  convertValuesForHtsViz() {
+  function convertValuesForHtsViz() {
```

Comment:
I'm thinking this function should be a separate hook that takes `eventDataRaw` as a parameter, then returns a memoized `eventDataProcessed`. In fact, we should probably be applying this approach to a lot of this code, since it feels like we're overusing `useEffect`, which will result in a lot of extra render cycles.

If everything works, feel free to implement these performance improvements later. A lot of these things weren't considered/possible when this component was a class component, so this work opens a lot of doors for improvement.

Reply:
One of the challenges I’m facing is infinite updates caused by multiple useEffects and their dependencies. When one useEffect updates, it triggers another, creating a loop. I’m working on finding a more efficient solution to this issue.

At first, I focused only on changing the syntax, but now I’m taking a step back to consider the entire scenario. I’m thinking about how I would approach the problem if I were developing it from scratch, making sure to avoid using class components.

Reply:
That's a good approach - this component is bloated and rethinking it from the ground up is what I would have done as well if I had more time.